### ISSUE TITLE
Virtual scroll - search does not work after keyboard navigation

### ISSUE THREAD

**Issue by paustint** [jetstreamapp/jetstream#205]
**Describe the bug**

1. keyboard navigation remembers prior selection when searching again
2. escape does not close menu
3. typing in a search term does not auto-open if closed

After keyboard navigation, the search does not work properly.
The focus keeps switching to the previously focused item every time search results change.

After each change in search, we need to make sure that the focused item is cleared.
